crypto/ed25519/ed25519.go

```diff
@@ -73,42 +92,57 @@ func (privKey PrivKeyEd25519) Equals(other crypto.PrivKey) bool {
 	}
 }
 
-func (privKey PrivKeyEd25519) ToCurve25519() *[32]byte {
+// ToCurve25519 takes a private key and returns its representation on
+// Curve25519. Curve25519 is birationally equivalent to Edwards25519,
+// which Ed25519 uses internally. This method is for intended use in
+// Diffie Hellman key exchanges. (i.e. X25519)
+func (privKey PrivKeyEd25519) ToCurve25519() *[PubKeyEd25519Size]byte {
 	keyCurve25519 := new([32]byte)
 	privKeyBytes := [64]byte(privKey)
 	extra25519.PrivateKeyToCurve25519(keyCurve25519, &privKeyBytes)
 	return keyCurve25519
 }
 
-// Deterministically generates new priv-key bytes from key.
+// Generate deterministically generates a new priv-key bytes from key.
+// The privkey is generated as Sha256(amino_encode({privkey, index}))
+// Note that we are using the non-standard golang definition for ed25519
+// privkeys that includes the pubkey appended to it
```

Comment:
"Note that similar to golang/x/crypto/ed25519 we include the private key ..."
Maybe, even add a link to: https://github.com/golang/crypto/blob/a49355c7e3f8fe157a85be2f77e6e269a0f89602/ed25519/ed25519.go#L48-L52 ?

The way we derive the key here is non-standard anyways?